packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -27,23 +27,33 @@ export async function makeWallet(
   const brandToIssuer = makeStore();
   const brandToMath = makeStore();
 
-  // OfferDescs that the wallet knows about (the inbox).
-  const idToOfferDesc = new Map();
+  // Offers that the wallet knows about (the inbox).
+  const idToOffer = makeStore();
 
-  // Compiled offerDescs (all ready to execute).
-  const idToCompiledOfferDescP = new Map();
+  // Compiled offers (all ready to execute).
+  const idToCompiledOfferP = new Map();
   const idToCancel = new Map();
 
   // Client-side representation of the purses inbox;
   const pursesState = new Map();
   const inboxState = new Map();
 
   function getPursesState() {
-    return JSON.stringify([...pursesState.values()]);
+    const entries = [...pursesState.entries()];
+    const values = entries
+      .sort(([id1], [id2]) => id1 > id2)
```

Comment:
Why do they need to be sorted? Is it worth adding a comment?

Reply:
Determinism.  I've added a comment.